File: app/src/main/java/top/txwgoogol/designpattern/mvp/WeatherActivity.java
package top.txwgoogol.designpattern.mvp;

import android.os.Bundle;
import android.widget.Button;
import android.widget.TextView;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;
import top.txwgoogol.designpattern.R;

import static com.bumptech.glide.util.Preconditions.checkNotNull;

/**
 * 主界面
 * Created by txwgoogol on 2018/2/13.
 */
public class WeatherActivity extends AppCompatActivity implements WeatherContact.View {

    @BindView(R.id.ok_http)
    Button okHttp;
    @BindView(R.id.tv)
    TextView tv;

    private ProgressDialogEx mProgressDialogEx;
    private WeatherContact.Presenter mPresenter;
    private WeatherPresenter mWeatherPresenter;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_weather);
        ButterKnife.bind(this);

        mProgressDialogEx = new ProgressDialogEx(this);
        mWeatherPresenter = new WeatherPresenter(this);
    }

    @OnClick(R.id.ok_http)
    public void onViewClicked() {
        mPresenter.start();
    }

    @Override
    public void loading() {
        mProgressDialogEx.showProgressDialog();
    }

    @Override
    public void hide() {
        mProgressDialogEx.dismissProgressDialog();
    }

    @Override
    public void success(Weather weather) {
        dataSuccess(weather);
    }

    @Override
    public void failure() {
        mProgressDialogEx.dismissProgressDialog();
    }

    @Override
    public void setPresenter(WeatherContact.Presenter presenter) {
        mPresenter = checkNotNull(presenter);
    }

    //填充数据
    private void dataSuccess(Weather model) {
        Weather.WeatherinfoBean weatherinfo = model.getWeatherinfo();
        String showData = getResources().getString(R.string.city) + weatherinfo.getCity()
                + getResources().getString(R.string.wd) + weatherinfo.getWD()
                + getResources().getString(R.string.ws) + weatherinfo.getWS()
                + getResources().getString(R.string.time) + weatherinfo.getTime();
        tv.setText(showData);
    }

}

File: app/src/main/java/top/txwgoogol/designpattern/mvp/WeatherActivity.kt
package top.txwgoogol.designpattern.mvp

import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.bumptech.glide.util.Preconditions.checkNotNull
import kotlinx.android.synthetic.main.activity_weather.*
import top.txwgoogol.designpattern.R

/**
 * 主界面
 * Created by txwgoogol on 2018/2/13.
 */
class WeatherActivity : AppCompatActivity(), WeatherContact.View, View.OnClickListener {
	
	private var mProgressDialogEx: ProgressDialogEx? = null
	private var mPresenter: WeatherContact.Presenter? = null
	private var mWeatherPresenter: WeatherPresenter? = null
	
	override fun onCreate(savedInstanceState: Bundle?) {
		super.onCreate(savedInstanceState)
		setContentView(R.layout.activity_weather)
		
		mProgressDialogEx = ProgressDialogEx(this)
		mWeatherPresenter = WeatherPresenter(this)
		
		ok_http.setOnClickListener(this)
	}
	
	override fun onClick(v: View) {
		when(v.id){
			R.id.ok_http -> mPresenter!!.start()
		}
	}
	
	override fun loading() {
		mProgressDialogEx!!.showProgressDialog()
	}
	
	override fun hide() {
		mProgressDialogEx!!.dismissProgressDialog()
	}
	
	override fun success(weather: Weather) {
		dataSuccess(weather)
	}
	
	override fun failure() {
		mProgressDialogEx!!.dismissProgressDialog()
	}
	
	override fun setPresenter(presenter: WeatherContact.Presenter) {
		mPresenter = checkNotNull<WeatherContact.Presenter>(presenter)
	}
	
	//填充数据
	private fun dataSuccess(model: Weather) {
		val weatherinfo = model.weatherinfo
		val showData = (resources.getString(R.string.city) + weatherinfo.city
			+ resources.getString(R.string.wd) + weatherinfo.wd
			+ resources.getString(R.string.ws) + weatherinfo.ws
			+ resources.getString(R.string.time) + weatherinfo.time)
		tv_desc.text = showData
	}
	
}